src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2241,14 +2261,32 @@ public static Map<String,String> getInvertiblePinMap(Series series, Unisim unisi
 	 * instances from which to copy the implementation. If targeting the top instance, use an 
 	 * empty String ("") as the destination instance name.
 	 */
-	public static void copyImplementation(Design src, Design dest, boolean lockPlacement, 
+	public static void copyImplementation(Design src, Design dest, boolean lockPlacement,
+										  boolean lockRouting, Map<String,String> srcToDestInstNames) {
+		copyImplementation(src, dest, false, false, lockPlacement, lockRouting, srcToDestInstNames);
+	}
+
+	/**
+	 * Copies the logic and implementation of a set of cells from one design to another with additional flags to control copying nets.
+	 * @param src The source design (with partial or full implementation)
+	 * @param dest The destination design (with matching cell instance interfaces)
+	 * @param copyStaticNets Flag indicating if static nets should be copied
+	 * @param copyOnlyInternalNets Flag indicating if only nets with every terminal inside the cell should be copied
+	 * @param lockPlacement Flag indicating if the destination implementation copy should have the
+	 * 	placement locked
+	 * @param lockRouting Flag indicating if the destination implementation copy should have the
+	 * 	routing locked
+	 * @param srcToDestInstNames A map of source (key) to destination (value) pairs of cell
+	 * instances from which to copy the implementation
+	 */
+	public static void copyImplementation(Design src, Design dest, boolean copyStaticNets, boolean copyOnlyInternalNets, boolean lockPlacement,
```

Comment:
Two new parameters have been added to this method: `copyOnlyInternalNets` and `copyStaticNets`. Are they both necessary?

Would it be better to pick a behaviour as default and stick with that?

Reply:
Yes, it it does not break any existing use cases.

Is there any existing code that use copyImplementation?  I presume yes, because copyImplementation added before this PR. The behavior of the existing copyImplementation is maintained with these two flags false.

Reply:
I would say we could make copying static nets the default.  However, I can see situations that might be useful where partial routing is copied.  So maybe keeping one of the flags would be useful.  Perhaps `copyOnlyInternalNets` could be inverted and called `copyPartialNetRouting`

Reply:
We should not rush to a not-urgent decision and should leave the default as it is. The default change should be made outside of this PR.

Reply:
> I would say we could make copying static nets the default. However, I can see situations that might be useful where partial routing is copied. So maybe keeping one of the flags would be useful. Perhaps `copyOnlyInternalNets` could be inverted and called `copyPartialNetRouting`
When copyOnlyInternalNets is true, only the net completely in the cell is copy, ie., no partial nets are copied.

Reply:
Leave as is.